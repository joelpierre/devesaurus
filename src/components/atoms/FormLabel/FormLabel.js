import React from 'react';
import PropTypes from 'prop-types';
import styles from './FormLabel.module.scss';

const FormLabel = ({ text, name }) => {
  return (
    <label
      id={name}
      data-test="component-label"
      htmlFor={name}
      aria-label={`Label for ${name}`}
      className={styles.form__label}
    >
      {text}
    </label>
  );
};

FormLabel.propTypes = {
  text: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default FormLabel;
