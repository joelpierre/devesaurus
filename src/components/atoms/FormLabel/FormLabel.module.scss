.form__label {
  padding: 0;
  line-height: 1.3;
  margin-bottom: 1rem;

  &--sm {
    font-size: 1.4rem;
  }

  &--md {
    font-size: 1.6rem;
  }

  &--lg {
    font-size: 1.8rem;
  }
}
