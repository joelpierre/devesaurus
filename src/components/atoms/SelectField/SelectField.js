import React from 'react';
import PropTypes from 'prop-types';
import styles from './SelectField.module.scss';

const SelectField = ({ name, placeholder, options }) => {
  return (
    <div data-test="component-select-field" className={styles['select-field']}>
      <select
        className={styles['select-field__select']}
        name={name}
        placeholder={placeholder}
      >
        {options.map((option, index) => {
          return (
            <option key={index} value={option.value}>
              {option.name}
            </option>
          );
        })}
      </select>
    </div>
  );
};

SelectField.defaultProps = {
  placeholder: null,
};

SelectField.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default SelectField;
