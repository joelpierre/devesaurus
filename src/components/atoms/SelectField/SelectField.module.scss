@import "~sass/core";

.select-field {
  position: relative;
  width: 100%;
  display: inline-block;

  &__select {
    display: inline-block;
    width: 100%;
    min-height: 45px;
    padding: 12px 16px;
    line-height: 1;
    color: $color-tint-omega;
    vertical-align: middle;
    background: $color-tint-alpha url("data:image/svg+xml;charset=utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'><path fill='%23343a40' d='M2 0L0 2h4zm0 5L0 3h4z'/></svg>") no-repeat right .75rem center;
    background-size: 8px 16px;
    border: 1px solid $color-mid-gray-1;
    border-radius: $radius-rounded;
    appearance: none;
  }
}
