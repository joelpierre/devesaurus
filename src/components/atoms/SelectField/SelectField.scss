.select-wrapper {
  position: relative;
  width: 100%;
}
