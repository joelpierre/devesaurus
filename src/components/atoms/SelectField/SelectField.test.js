import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

import SelectField from './SelectField';
import FormLabel from '../FormLabel/FormLabel';

const defaultProps = {
  name: 'test-name',
  options: [
    {
      name: 'Test Option Name',
      value: 'test-option-value',
    },
  ],
};

/**
 * Factory function to create a ShallowWrapper for the SelectField component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<SelectField {...setupProps} />);
};

describe('<SelectField/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the SelectField Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-select-field');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(SelectField, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should render with FormLabel', () => {
    wrapper = setup({
      ...defaultProps,
      label: 'hello',
    });

    expect(wrapper.find(FormLabel)).toBeTruthy();
    matchSnapshot(wrapper);
  });
});
