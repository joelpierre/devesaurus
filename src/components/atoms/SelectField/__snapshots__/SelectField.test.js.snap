// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectField/> should match snapshot 1`] = `
<Fragment>
  <div
    className="select-field"
    data-test="component-select-field"
  >
    <select
      className="select-field__select"
      name="test-name"
      placeholder={null}
    >
      <option
        key="0"
        value="test-option-value"
      >
        Test Option Name
      </option>
    </select>
  </div>
</Fragment>
`;

exports[`<SelectField/> should render with FormLabel 1`] = `
<Fragment>
  <FormLabel
    name="test-name"
    text="hello"
  />
  <div
    className="select-field"
    data-test="component-select-field"
  >
    <select
      className="select-field__select"
      name="test-name"
      placeholder={null}
    >
      <option
        key="0"
        value="test-option-value"
      >
        Test Option Name
      </option>
    </select>
  </div>
</Fragment>
`;
